spec/std/concurrent_spec.cr

```diff
@@ -8,38 +8,65 @@ private def method_named(expected_named)
   Fiber.current.name.should eq(expected_named)
 end
 
+class SomeParallelJobException < Exception
+end
+
+private def raising_job : String
+  raise SomeParallelJobException.new("boom")
+  "result"
+end
+
 describe "concurrent" do
-  it "does four things concurrently" do
-    a, b, c, d = parallel(1 + 2, "hello".size, [1, 2, 3, 4].size, nil)
-    a.should eq(3)
-    b.should eq(5)
-    c.should eq(4)
-    d.should be_nil
-  end
+  describe "parallel" do
+    it "does four things concurrently" do
+      a, b, c, d = parallel(1 + 2, "hello".size, [1, 2, 3, 4].size, nil)
+      a.should eq(3)
+      b.should eq(5)
+      c.should eq(4)
+      d.should be_nil
+    end
 
-  it "uses spawn macro" do
-    chan = Channel(Int32).new
+    it "re-raises errors from Fibers as ConcurrentExecutionException" do
+      exception = expect_raises(ConcurrentExecutionException) do
+        a, b = parallel(raising_job, raising_job)
+      end
 
-    spawn method_with_named_args(chan)
-    chan.receive.should eq(3)
+      exception.inspect_with_backtrace.should contain("Caused by: boom (SomeParallelJobException)")
```

Comment:
Better not rely on the backtrace format, use `exception.cause.should be_a(SomeParallelJobException)` instead?

Reply:
makes sense, updated